yaml/_dumper_state.ts

```diff
@@ -766,39 +758,38 @@ export class DumperState {
       _result += pairBuffer;
     }
 
-    this.tag = _tag;
     this.dump = _result || "{}"; // Empty mapping if no valid pairs.
   }
 
-  detectType(explicit: boolean): boolean {
+  detectType(explicit: boolean) {
```

Comment:
The return type for this method has changed from explicitly `boolean` to now being implicitly `string | null`. Why? Return types should be explicit and not change.

Reply:
Since the `tag` property is removed and was set inside this function, it has to be returned instead. It returns `string`(former `true`) and `null` (former `false`).